WhistleBlowerClient/app/src/main/java/com/android/example/myapplication/GroupsActivity.java

```diff
@@ -0,0 +1,64 @@
+package com.android.example.myapplication;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+import java.util.ArrayList;
+
+public class GroupsActivity extends AppCompatActivity {
+    private RecyclerView recyclerView;
+    private RecyclerView.Adapter adapter; // the bridge between the list of the groups and the Recycler view
+    private RecyclerView.LayoutManager layoutManager; // for aligning group's items in our list (creates the list view)
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
very long function

Reply:
it's too long because i've added some testing examples of groups, but yeah you are right.
i will move the testing examples to a separate function.